src/utils/validateEventIds.ts

```diff
@@ -0,0 +1,10 @@
+import { validate as validateUUID } from "uuid";
+
+import type { EventData, EventType } from "../types";
+import { IsNotUUIDError } from "./IsNotUUIDError";
+
+export const validateEventIds = (events: EventData<EventType>[]): void => {
+  events.forEach((event) => {
+    if (!validateUUID(event.id)) throw new IsNotUUIDError(event.id);
+  });
+};
```

Comment:
Thank you, @Imanghvs. We should consider utilizing a validation library, like zod, to validate the inputs users provide to functions and to improve error messages, particularly if they're not using TypeScript. We are conducting some experiments to determine if zod is a good fit for us. If we decide to go that route, you can implement the necessary changes if you are still willing to contribute :)

Reply:
I absolutely like to contribute. Also I believe `zod` is a great choice (better than `class-validator` for this project). Please let me know your decision and I'll go for it. :)

Reply:
We have decided to proceed with Zod, but we will address it incrementally. I will close this pull request for the time being.